bot/exts/utilities/color.py

```diff
@@ -0,0 +1,337 @@
+import colorsys
+import json
+import logging
+import re
+from io import BytesIO
+
+from PIL import Image, ImageColor
+from discord import Embed, File
+from discord.ext import commands
+from rapidfuzz import process
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+
+logger = logging.getLogger(__name__)
+
+
+ERROR_MSG = """The color code {user_color} is not a possible color combination.
+\nThe range of possible values are:
+\nRGB & HSV: 0-255
+\nCMYK: 0-100%
+\nHSL: 0-360 degrees
+\nHex: #000000-#FFFFFF
+"""
+
+COLOR_JSON_PATH = "bot/resources/utilities/ryanzec_colours.json"
+with open(COLOR_JSON_PATH) as f:
+    COLOR_MAPPING = json.load(f)
+
+
+class Color(commands.Cog):
+    """User initiated commands to receive color information."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command(aliases=["colour"])
+    async def color(self, ctx: commands.Context, mode: str, *, user_color: str) -> None:
+        """
+        Send information on input color code or color name.
+
+        Possible modes are: "hex", "rgb", "hsv", "hsl", "cmyk" or "name".
+        """
+        logger.debug(f"{mode = }")
+        logger.debug(f"{user_color = }")
+        if mode.lower() == "hex":
+            await self.hex_to_rgb(ctx, user_color)
+        elif mode.lower() == "rgb":
+            rgb_color = self.tuple_create(user_color)
+            await self.color_embed(ctx, rgb_color)
+        elif mode.lower() == "hsv":
+            await self.hsv_to_rgb(ctx, user_color)
+        elif mode.lower() == "hsl":
+            await self.hsl_to_rgb(ctx, user_color)
+        elif mode.lower() == "cmyk":
+            await self.cmyk_to_rgb(ctx, user_color)
+        elif mode.lower() == "name":
+            color_name, hex_color = self.match_color_name(user_color)
+            if "#" in hex_color:
+                rgb_color = ImageColor.getcolor(hex_color, "RGB")
+            else:
+                rgb_color = ImageColor.getcolor("#" + hex_color, "RGB")
+            await self.color_embed(ctx, rgb_color, color_name)
+        else:
+            # mode is either None or an invalid code
+            if mode is None:
+                no_mode_embed = Embed(
+                    title="No 'mode' was passed, please define a color code.",
```

Comment:
No need for the quotes, we're referring to `mode` as a mode. Not *specifically* as an argument
```suggestion
                    title="No mode was passed, please define a color code.",
```